### PR TITLE
Fix Build and Get BSAP to run

### PR DESCRIPTION
**Pull request by CodeforEvolution**
Hello! I have fixed BSAP and it now runs! It will open the dictionary files included in the lib directory and is quite stable!

These ToDos are beyond what the task requires but I caught will working:
TODO:
Fix sqlite engine: 
The sqlite engine seems to be the reason for the crash on start up and besides needs to be updated to use sqlite3 as it currently uses the way obsolete sqlite2

Fix gcc2 build:
It will build on the gcc2_x86 hybrid, though only after typing "setarch x86",
basically, builds with gcc5, not gcc2

Bug:
The program will crash if you switch to the Ydp dictionary engine and cancel out without choosing a valid directory with the files.

### CODE REVIEW

**Comment by ytmytm** [comment]
Thank you for your work!
